packaging/kibana_plugin.py

```diff
@@ -0,0 +1,231 @@
+    # remove es- prefix
+    for string in ("elasticsearch-", "es-"):
+        if repo.startswith(string):
+            return repo[len(string):]
+
+    return repo
+
+def is_plugin_present(plugin_dir, working_dir):
+    return os.path.isdir(os.path.join(working_dir, plugin_dir))
+
+def parse_error(string):
+    reason = "reason: "
+    try:
+        return string[string.index(reason) + len(reason):].strip()
+    except ValueError:
+        return string
+
+def install_plugin(module, plugin_bin, plugin_name, url, timeout):
+    cmd_args = [plugin_bin, "plugin", PACKAGE_STATE_MAP["present"], plugin_name]
+
+    if url:
+        cmd_args.append("--url %s" % url)
+
+    if timeout:
+        cmd_args.append("--timeout %s" % timeout)
+
+    cmd = " ".join(cmd_args)
+
+    rc, out, err = module.run_command(cmd)
+
+    if rc != 0:
+        reason = parse_error(out)
+        module.fail_json(msg=reason)
+    
+    return True, cmd, out, err
+
+def remove_plugin(module, plugin_bin, plugin_name):
+    cmd_args = [plugin_bin, "plugin", PACKAGE_STATE_MAP["absent"], plugin_name]
+
+    cmd = " ".join(cmd_args)
+
+    rc, out, err = module.run_command(cmd)
+
+    if rc != 0:
+        reason = parse_error(out)
+        module.fail_json(msg=reason)
+    
+    return True, cmd, out, err
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            state=dict(default="present", choices=PACKAGE_STATE_MAP.keys()),
+            url=dict(default=None),
+            timeout=dict(default="1m"),
+            plugin_bin=dict(default="/opt/kibana/bin/kibana", type="path"),
+            plugin_dir=dict(default="/opt/kibana/installedPlugins/", type="path"),
+            version=dict(default=None),
+            force=dict(default="no", type="bool")
+        )
+    )
+
+    name        = module.params["name"]
+    state       = module.params["state"]
+    url         = module.params["url"]
+    timeout     = module.params["timeout"]
+    plugin_bin  = module.params["plugin_bin"]
+    plugin_dir  = module.params["plugin_dir"]
+    version     = module.params["version"]
+    force       = module.params["force"]
+
+    present = is_plugin_present(parse_plugin_repo(name), plugin_dir)
+
+    # skip if the state is correct
+    if (present and state == "present" and not force) or (state == "absent" and not present and not force):
+        module.exit_json(changed=False, name=name)
+
+    if (version):
+       name = name + '/' + version
+
+    if state == "present":
+        if force:
+            remove_plugin(module, plugin_bin, name)
+        changed, cmd, out, err = install_plugin(module, plugin_bin, name, url, timeout)
+
+    elif state == "absent":
+        changed, cmd, out, err = remove_plugin(module, plugin_bin, name)
+
+    module.exit_json(changed=changed, cmd=cmd, name=name, state=state, url=url, timeout=timeout, stdout=out, stderr=err)
+
+from ansible.module_utils.basic import *
+
+main()
```

Comment:
The call to main() needs to check __name__ == __main__ first.

See http://docs.ansible.com/ansible/developing_modules.html#module-checklist 

> Call your main() from a conditional so that it would be possible to import them into unittests in the future example:
> 
> if __name__ == '__main__':
>     main()